src/haven/OverviewTool.java

```diff
@@ -90,10 +90,13 @@ private void update_text()
         
         int height = 25;
         //destroy all previous labels
-        for(Label l : ls)
-        {
-            l.destroy();
-        }
+		//your iteration method was conflicting with on-the-go destroying, try or fix this method
+		Iterator<Label> itr = ls.iterator();
+		while(itr.hasNext()) {
+			Label l = itr.next();
+			itr.remove()
```

Comment:
Not really >.> bad practice I know, was meant more as a bug report with a halfassed suggestion at a solution, but was hoping kand would have a look at it to fix it properly :p

Sent from my Windows Phone
________________________________
From: LOMS<mailto:notifications@github.com>
Sent: ‎16/‎07/‎2015 22:33
To: DonnEssime/Custom-Salem<mailto:Custom-Salem@noreply.github.com>
Cc: lachlaan<mailto:lachlaan@hotmail.com>
Subject: Re: [Custom-Salem] Fixed crash-inducing iterator drive-by-destruction. (#5)

> @@ -90,10 +90,13 @@ private void update_text()
>
>          int height = 25;
>          //destroy all previous labels
> -        for(Label l : ls)
> -        {
> -            l.destroy();
> -        }
> +             //your iteration method was conflicting with on-the-go destroying, try or fix this method
> +             Iterator<Label> itr = ls.iterator();
> +             while(itr.hasNext()) {
> +                     Label l = itr.next();
> +                     itr.remove()

Have you tried to compile this?

---
Reply to this email directly or view it on GitHub:
https://github.com/DonnEssime/Custom-Salem/pull/5/files#r34834589